### PR TITLE
feat(chat): add quick app editor (Issue #2160)

### PR DESCRIPTION
**Pull request by Gimir**
**Description:**

- add monaco-editor
- add quick app wizard
- add model description parsers for extracting and packing quick app config

Issues:

- Issue #2160

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2277.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Gimir** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2277.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/utils/app/application.ts [hunk 1]
```diff
@@ -105,3 +107,48 @@ export const convertApplicationFromApi = (
...
+  return !!description?.includes(QUICK_APP_CONFIG_DIVIDER);
+};
+
+export const getModelDescription = (entity: DialAIEntityModel) => {
+  const [description] =
+    entity.description && isQuickApp(entity)
+      ? entity.description.split(QUICK_APP_CONFIG_DIVIDER)
+      : [entity.description];
+
+  return description;
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/application.ts]:
```suggestion
  const [description] = (entity.description ?? '').split(QUICK_APP_CONFIG_DIVIDER);

  return description;
```

File: apps/chat/src/components/Chat/ModelDescription.tsx [hunk 2]
```diff
@@ -35,13 +36,13 @@ export const ModelDescription = ({
...
       </div>
-      {model.description && (
+      {!!getModelDescription(model) && (
         <span
           className="whitespace-pre-wrap text-xs text-secondary"
           data-qa="entity-descr"
         >
           <EntityMarkdownDescription isShortDescription={isShortDescription}>
-            {model.description}
+            {getModelDescription(model) as string}
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ModelDescription.tsx]: `getModelDescription` should always return `string`

File: apps/chat/src/utils/app/application.ts [hunk 3]
```diff
@@ -105,3 +107,48 @@ export const convertApplicationFromApi = (
...
+};
+
+export const createQuickAppConfig = ({
+  description,
+  config,
+}: {
+  description: string;
+  config: string;
+}) => {
+  return `${description.trim()}${QUICK_APP_CONFIG_DIVIDER}${config.trim()}`;
```
Comment (IlyaBondar) [inline apps/chat/src/utils/app/application.ts]:
```suggestion
  return [description.trim(), config.trim()].join(QUICK_APP_CONFIG_DIVIDER);
```

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2277.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by IlyaBondar — APPROVED** [review]

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2277.nightly-test.deltixhub.io
>E2E tests status: success

**Pull request merged**